src/Types/ArrayType.php

```diff
@@ -27,19 +27,13 @@ public function getSQLDeclaration(array $column, AbstractPlatform $platform): st
         return $platform->getClobTypeDeclarationSQL($column);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function convertToDatabaseValue($value, AbstractPlatform $platform)
+    public function convertToDatabaseValue(mixed $value, AbstractPlatform $platform): ?string
     {
         // @todo 3.0 - $value === null check to save real NULL in database
         return serialize($value);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function convertToPHPValue($value, AbstractPlatform $platform)
+    public function convertToPHPValue(mixed $value, AbstractPlatform $platform): mixed
     {
         if ($value === null) {
```

Comment:
This is another pet peeve of mine: every type has to check if `$value` is NULL prior to the conversion. I'm curious if we should move this check one level up and only expect the non-NULL values in the converters. This is just food for thought, not something to be addressed in this PR.